File: microservices/order-approval/src/eureka.js
// import * as dotenv from 'dotenv';
// import { Eureka } from 'eureka-js-client';
const dotenv = require('dotenv')
const { Eureka } = require('eureka-js-client')
const EnvError = require('./errors/EnvError');
dotenv.config();

const APPLICATION_HOST = process.env.HOST;
const APPLICATION_PORT = Number(process.env.PORT);
const APPLICATION_NAME = process.env.NAME;
const EUREKA_HOST = process.env.EUREKA_HOST;
const EUREKA_PORT = Number(process.env.EUREKA_PORT);

if (APPLICATION_HOST == undefined) throw new EnvError('HOST');
if (APPLICATION_PORT == undefined) throw new EnvError('PORT');
if (APPLICATION_NAME == undefined) throw new EnvError('NAME');
if (EUREKA_HOST == undefined) throw new EnvError('EUREKA_HOST');
if (EUREKA_PORT == undefined) throw new EnvError('EUREKA_PORT');

const client = new Eureka({
	instance: {
		app: APPLICATION_NAME,
		hostName: APPLICATION_HOST,
		ipAddr: APPLICATION_HOST,
		port: { $: APPLICATION_PORT, '@enabled': true },
		vipAddress: 'www.express-server.com',
		dataCenterInfo: { '@class': 'com.netflix.appinfo.InstanceInfo$DefaultDataCenterInfo', name: 'MyOwn' },
	},
	eureka: { host: EUREKA_HOST, port: EUREKA_PORT, servicePath: '/eureka/apps/' },
});

module.exports = client;

File: microservices/product-service/src/eureka.py
import requests
import json
import os

from dotenv import load_dotenv
load_dotenv(verbose=True)

APP_NAME = os.getenv("APP_NAME")
APP_PORT = os.getenv("PORT")
HOST = os.getenv("HOST")

EUREKA_HOST = os.getenv("EUREKA_HOST")
EUREKA_PORT = os.getenv("EUREKA_PORT")

EUREKA_SERVER = f'http://{EUREKA_HOST}:{EUREKA_PORT}/eureka/apps/{APP_NAME}'

payload = {
    'instance': {
        'hostName': HOST,
        'app': APP_NAME,
        'ipAddr': HOST,
        'port': {'$': APP_PORT, '@enabled': True},
        'dataCenterInfo': {'@class': 'com.netflix.appinfo.MyDataCenterInfo', 'name': 'MyOwn'}
    }
}

headers = {'content-type': 'application/json', 'cache-control': 'no-cache'}
json_payload = json.dumps(payload)


def start():
    response = requests.request('POST', EUREKA_SERVER, data=json_payload, headers=headers)
    print('status code::', response.status_code)
